socket_linux.go

```diff
@@ -18,11 +18,11 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 	// unix.SO_REUSEPORT is not defined on linux 386/amd64, see
 	// https://github.com/golang/go/issues/16075
 	if reuseport {
-		if err := unix.SetsockoptInt(sockFD, unix.SOL_SOCKET, 0xf, 1); err != nil {
+		if err = unix.SetsockoptInt(sockFD, unix.SOL_SOCKET, 0xf, 1); err != nil {
```

Comment:
It was a warning in the editor about shadowing the definition of err, which I realize is a thing that happens all the time. Happy to remove.